pandas/core/arrays/base.py

```diff
@@ -653,15 +655,73 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __repr__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'{class_name}'
+            u'{data}\n'
+            u'Length: {length}, dtype: {dtype}'
+        )
+        # the short repr has no trailing newline, while the truncated
+        # repr does. So we include a newline in our template, and strip
+        # any trailing newlines from format_object_summary
+        data = format_object_summary(self, self._formatter(), name=False,
+                                     trailing_comma=False).rstrip()
+        class_name = u'<{}>\n'.format(self.__class__.__name__)
+        return template.format(class_name=class_name, data=data,
+                               length=len(self),
+                               dtype=self.dtype)
+
+    def _formatter(self, formatter=None):
+        # type: (Optional[ExtensionArrayFormatter]) -> Callable[[Any], str]
+        """Formatting function for scalar values.
+
+        This is used in the default '__repr__'. The formatting function
+        receives instances of your scalar type.
+
+        Parameters
+        ----------
+        formatter: GenericArrayFormatter, optional
+            The formatter this array is being rendered with. When the array
+            is being rendered inside an Index, Series, or DataFrame, a
+            formatter will be provided. So if you want your objects to
+            render differently inside a Series from on its own, checking
+            with ``formatter is None`` is an option.
+
+            The default behavior depends on whether `formatter` is passed.
+
+            * When `formatter` is None, :func:`repr` is returned.
+            * When `formatter` is passed, ``formatter.formatter`` is used,
```

Comment:
But what is ``formatter.formatter`` doing? 
(from the viewpoint of an EA author, I want to know what the default would be for my EA)